src/bag/interface/skill.py

```diff
@@ -583,6 +586,8 @@ def write_tech_info(self, tech_name: str, out_yaml: str) -> None:
         lay_dict = {}
         via_keys = list(via_id_dict.keys())
         for sub_str in all_lay_str.split(') ('):
+            if not sub_str:  # blank edge case
+                continue 
```

Comment:
I think the reason for this error is that the derived layers string is empty. Instead of catching the error here, I recommend checking if the derived layers string has any content and create all_lay_str accordingly.